### DISCUSSION TITLE
2D hand pose by YOLOX-POSE

### DISCUSSION THREAD

**Discussion by ghost** [open-mmlab/mmpose#2235] [Q&A]
Hi, there.  I wanna to train 2D hand keypoints of coco wholebody hand by YOLOX-POSE.
Is this possible??
 I tried to edit this [config file](https://github.com/open-mmlab/mmpose/blob/main/projects/yolox-pose/configs/yolox-pose_s_8xb32-300e_coco.py).  But it was difficult to me.

**Comment by ghost** [comment]
![Screenshot from 2023-04-15 17-48-25](https://user-images.githubusercontent.com/126233717/232202432-082509d5-9189-47aa-97ef-ec728ab97bb7.png)

**Reply by Ben-Louis** [reply]
The `metainfo` argument in the config file should be modified to the corresponding one for the dataset you are using

**Reply by ghost** [reply]
Thank you for reply. I edited. bun it wasn't run. 
Could you please help me?

![Screenshot from 2023-04-16 20-18-24](https://user-images.githubusercontent.com/126233717/232306769-3cf383f5-2975-487a-8ffd-30e0b2aa12f6.png)
![Screenshot from 2023-04-16 20-22-56](https://user-images.githubusercontent.com/126233717/232306795-7ae29d70-4230-4eb1-b2f1-1a94b17fdedc.png)

![Screenshot from 2023-04-16 20-23-19](https://user-images.githubusercontent.com/126233717/232306817-e077f1c6-44db-425a-bc7f-da645bcbc735.png)

**Reply by Ben-Louis** [reply]
To better assist you in resolving the issue, it would be beneficial if you could provide the complete contents of your configuration file.

**Reply by ghost** [reply]
[This](https://github.com/keitatoyota/mmpose/blob/main/projects/yolox-pose/configs/my_config_hand_yolox-pose_s_8xb32-300e_coco.py) is config file I edited.

**Reply by Ben-Louis** [reply]
Please refer to https://github.com/open-mmlab/mmpose/blob/8dd07a75eb48eb0d482c73cf70eaa03e0a98767e/configs/hand_2d_keypoint/topdown_heatmap/coco_wholebody_hand/td-hm_hrnetv2-w18_dark-8xb32-210e_coco-wholebody-hand-256x256.py#L98-L158 to modify the arguments `dataset_type` and `val_evaluator`

**Reply by Ben-Louis** [reply]
Simultaneously, you may need to add the following code in the `$MMPOSE/projects/yolox-pose/datasets/coco_dataset.py`:
```python
from mmpose.datasets import CocoWholeBodyHandDataset as MMPoseCocoWholeBodyHandDataset

@DATASETS.register_module()
class CocoWholeBodyHandDataset(MMPoseCocoWholeBodyHandDataset):

    @force_full_init
    def prepare_data(self, idx) -> Any:
        data_info = self.get_data_info(idx)
        data_info['dataset'] = self
        return self.pipeline(data_info)
```

**Reply by ghost** [reply]
Thank you for always. 
I have to do other projects now. So I'll try them weekend.
